Library/Homebrew/cmd/test-bot.rb

```diff
@@ -287,6 +298,9 @@ def brew_update
       if ENV["ghprbPullLink"]
         @url = ENV["ghprbPullLink"]
         @hash = nil
+      elsif ENV["TRAVIS_PULL_REQUEST"] && ENV["TRAVIS_PULL_REQUEST"] != "false"
+        @url = "https://github.com/#{ENV["TRAVIS_REPO_SLUG"]}/pull/#{ENV["TRAVIS_PULL_REQUEST"]}"
+        @hash = nil
```

Comment:
To secure the token, you need to store them somewhere else and make sure the code is read-only. Hence, we need dependent build.